docs/configuration.rst

```diff
@@ -1,5 +1,250 @@
+
+SERVER_OPTIONS
+^^^^^^^^^^^^^^
+
+This setting defines a list of key-value pairs specifying options for the gRPC server. These options help configure server behavior, such as setting limits on the size of incoming or outgoing messages.
+
+Example:
+
+If you want to set the maximum size for sending and receiving messages to 100MB, you can configure the SERVER_OPTIONS as:
+
+.. code-block:: python
+
+  "SERVER_OPTIONS": [
+    ("grpc.max_send_message_length", 100 * 1024 * 1024),
+    ("grpc.max_receive_message_length", 100 * 1024 * 1024),
+  ],
+
+The above configuration allows the gRPC server to send and receive messages up to a size of 100MB.
+
+For more options, see the `documentation <https://grpc.github.io/grpc/core/group__grpc__arg__keys.html>`_.
+
+
+DEFAULT_AUTHENTICATION_CLASSES
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Defines the list of authentication classes the gRPC server uses to validate incoming requests. Requests are authenticated based on the methods provided by these classes, in the order they are listed.
+
+Example:
+
+If you want to set the maximum size for sending and receiving messages to 100MB, you can configure the SERVER_OPTIONS as:
+
+.. code-block:: python
+
+  "DEFAULT_AUTHENTICATION_CLASSES": [
+    "your_project.auth.JWTAuthentication"
+  ]
+
+
+DEFAULT_FILTER_BACKENDS
+^^^^^^^^^^^^^^^^^^^^^^^
+
+This setting designates the default filtering backends that gRPC services should use. Filtering backends allow requests to be filtered based on query parameters.
+
+For instance, to use Django's built-in filter backend:
+
+.. code-block:: python
+
+  "DEFAULT_FILTER_BACKENDS": ["django_filters.rest_framework.DjangoFilterBackend"]
+
+
+DEFAULT_PAGINATION_CLASS
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Defines the default pagination class for gRPC services. This class will be used to paginate large datasets in the response.
+
+Example configuration to use the `StandardResultsSetPagination` class:
+
+.. code-block:: python
+
+  "DEFAULT_PAGINATION_CLASS": "core.pagination.StandardResultsSetPagination"
+
+
+DEFAULT_PERMISSION_CLASSES
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This setting defines the list of default authentication classes that will be used for gRPC services. Each class specified in this list will be responsible for verifying the identity of the user making the request.
+
+For a hypothetical project that uses JWT for authentication:
+
+.. code-block:: python
+
+  "DEFAULT_PERMISSION_CLASSES": [
+      "your_project.permissions.IsAuthenticated",
+      "your_project.permissions.HasServiceAccess",
+  ]
+
+GRPC_ASYNC
+^^^^^^^^^^
+
+This setting determines the running mode of the gRPC server. If set to `True`, the server will operate in asynchronous mode. When in asynchronous mode, the server is capable of handling multiple concurrent requests using Python's `asyncio`. 
+
```

Comment:
Add that this setting is overriden to True when running the app with `grpcrunaioserver`